desktop-widgets/downloadfromdivecomputer.cpp

```diff
@@ -639,10 +640,12 @@ void DownloadFromDCWidget::enableBluetoothMode(int state)
 {
 	ui.chooseBluetoothDevice->setEnabled(state == Qt::Checked);
 
-	if (state == Qt::Checked)
-		selectRemoteBluetoothDevice();
-	else
+	if (state == Qt::Checked) {
+		if (ui.device->currentText().isEmpty()) 
```

Comment:
I think this will work in an undesirable way if a serial port is selected for the device, and the user then checks the bluetooth box - in this case the serial port device selection will be retained. I think a better way to do this will be to use `isBluetoothAddress()` here to check if the device is a bluetooth device.

Reply:
Thanks - that's better than just checking for a device id.